ios/VydiaRNFileUploader.swift

```diff
@@ -0,0 +1,537 @@
+//
+//  VydiaRNFileUploader.swift
+//  VydiaRNFileUploader
+//
+//  Created by Michael Czerniakowski on 19/05/2023.
+//  Copyright © 2023 Marc Shilling. All rights reserved.
+//
+
+import Foundation
+import Photos
+import MobileCoreServices
+
+@available(iOS 12, *)
+@objc(VydiaRNFileUploader)
+@objcMembers
+public class VydiaRNFileUploader: RCTEventEmitter {
+    
+    @objc public static var emitter: RCTEventEmitter?
+    
+    var filesMap: [String: URL] = [:]
+    var responsesData: [Int: NSMutableData] = [:]
+    var urlSession: URLSession?
+    var fileManager: FileManager
+    static var uploadId: Int = 0
+    static let BACKGROUND_SESSION_ID: String = "ReactNativeBackgroundUpload"
+    private static let synchronizationQueue = DispatchQueue(label: "com.example.synchronization")
+    
+    private static func getNextUploadId() -> Int {
+        defer {
+            uploadId += 1
+        }
+        return uploadId
+    }
+    
+    @objc override init() { //WIP!!!
+        self.fileManager = FileManager.default
+        super.init()
+        VydiaRNFileUploader.emitter = self
+    }
+    
+    override public static func requiresMainQueueSetup() -> Bool {
+        return true
+    }
+    
+    override public func sendEvent(withName eventName: String, body: Any?) {
+        if let emitter = VydiaRNFileUploader.emitter {
+//            emitter.sendEvent(withName: eventName, body: body)
```

Comment:
Should this be commented?

Reply:
Of course not :) RN I am dealing with some crashes. Upload itself worked ok without emitter, it goes berserk with emitter... 
<img width="1281" alt="Screenshot 2023-06-06 at 17 56 26" src="https://github.com/flip-shop/react-native-background-upload/assets/13233783/81ab70db-17ec-4de6-8a91-c79b03c53076">

<img width="1340" alt="Screenshot 2023-06-06 at 18 00 05" src="https://github.com/flip-shop/react-native-background-upload/assets/13233783/0ffa4978-a373-4f11-b6ae-1f6016d5f0c4">